sotopia/envs/multi_agent_parallel.py

```diff
@@ -0,0 +1,380 @@
+"""
+Multi-agent extension of ParallelSotopiaEnv to support more than 2 agents.
+"""
+
+import asyncio
+import copy
+import itertools
+from typing import Any, Optional
+
+try:
+    from beartype import beartype  # type: ignore[import-not-found]
+except ImportError:
+    # Fallback if beartype is not available
+    def beartype(func):  # type: ignore[no-untyped-def]
+        return func
+
+
+from sotopia.envs.parallel import (
+    ParallelSotopiaEnv,
+    get_bio,
+    render_text_for_environment,
+    render_text_for_agent,
+    _actions_to_natural_language,
+)
+from sotopia.messages.message_classes import MultiAgentBackground, ScriptBackground
+from sotopia.messages import MessengerMixin, Observation, AgentAction, SimpleMessage
+from sotopia.envs.evaluators import unweighted_aggregate_evaluate
+
+
+class MultiAgentSotopiaEnv(ParallelSotopiaEnv):
```

Comment:
Same here, let's just merge ParallelSotopiaEnv and MultiAgentSotopiaEnv together? If there's logic that we think needs separation, we could do that. However, instead of rewriting the function, we should do super.<function> etc.